src/rpc/src/batchQuery.ts

```diff
@@ -0,0 +1,56 @@
+import { AgoricChainStoragePathKind } from './types';
+import type { Unserialize } from '@endo/marshal';
+
+export const batchVstorageQuery = (
+  node: string,
+  unserialize: Unserialize<string>,
+  paths: [AgoricChainStoragePathKind, string][],
+) => {
+  const options = {
+    method: 'POST',
+    body: JSON.stringify(
+      paths.map(path => ({
+        jsonrpc: '2.0',
+        id: 1,
+        method: 'abci_query',
+        params: { path: `/custom/vstorage/${path[0]}/${path[1]}` },
+      })),
+    ),
+  };
+
+  return fetch(node, options)
+    .then(res => res.json())
+    .then(res =>
+      Object.fromEntries(
+        (Array.isArray(res) ? res : [res]).map((entry, index) => {
+          if (!entry.result.response.value) {
+            throw new Error(
+              'Cannot parse value of response for path' +
+                paths[index] +
+                ': ' +
+                entry,
+            );
+          }
+
+          const data = JSON.parse(atob(entry.result.response.value));
```

Comment:
running in Node this will import Node's `atob` which is deprecated. To clarify and avoid that warning please use,
```suggestion
          const data = JSON.parse(window.atob(entry.result.response.value));
```